app/Models/Beatmapset.php

```diff
@@ -424,6 +425,7 @@ public static function search(array $params = [])
             $ids = implode(',', $beatmap_ids);
             $beatmaps = static
                 ::with('beatmaps')
+                ->with('favorites')
```

Comment:
Not sure why the previous comment is gone but here it is again.

This doesn't help the `hasFavorite` method because the method doesn't use `$this->favorites` but a different query. And probably loads a bit too many data this way anyway.